mempool/cat/peers.go

```diff
@@ -0,0 +1,115 @@
+package cat
+
+import (
+	"fmt"
+
+	tmsync "github.com/tendermint/tendermint/libs/sync"
+	"github.com/tendermint/tendermint/mempool"
+	"github.com/tendermint/tendermint/p2p"
+)
+
+const firstPeerID = 1
+
+// mempoolIDs is a thread-safe map of peer IDs to short IDs used for tracking what peers have sent what
+// NOTE: taken from mempool/v1/reactor.go
+type mempoolIDs struct {
+	mtx       tmsync.RWMutex
+	peerMap   map[p2p.ID]uint16
```

Comment:
It would be beneficial to include an explanation of the meaning of the `uint16` values in `peerMap`, such as the order of insertion of IDs or the index of each ID, as it is not immediately clear.

Reply:
I can add a comment. I think the entire purpose of the `mempoolDs` struct is to have a more compact representation of the peers someone is connected to.